source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -533,12 +491,134 @@ namespace AdaptiveCards::Rendering::Uwp
+                                                                    winrt::ImageSource const& imageSource) {
+        if (isImageSvg)
+        {
+            // If we have an SVG, we need to try to parse for the image size before setting the image source
+            auto streamSize = stream.Size();
+            auto inputStream = stream.GetInputStreamAt(0);
+            auto streamDataReader = winrt::DataReader(inputStream);
+            auto loadDataReaderOperation = streamDataReader.LoadAsync((uint32_t)streamSize);
+
+            loadDataReaderOperation.Completed(
+                [weakThis = this->get_weak(), streamDataReader, uiElement, isAutoSize, parentElement, imageContainer, isVisible, stream, imageSource](
+                    winrt::IAsyncOperation<uint32_t> const& result, winrt::AsyncStatus status) -> void
+                {
+                    if (auto strongThis = weakThis.get())
+                    {
+                        if (status == winrt::AsyncStatus::Completed)
+                        {
+                            auto bytes = result.GetResults();
+                            auto svgText = streamDataReader.ReadString(bytes);
+
+                            auto svgDocument = winrt::XmlDocument();
+                            svgDocument.LoadXml(svgText);
+
+                            strongThis->ParseXmlForHeightAndWidth(svgDocument, imageSource);
+                        }
+
+                        // Now that we've parsed the height, we can set the image source
+                        strongThis->SetSvgImageSourceAsync(
+                            imageSource.as<winrt::SvgImageSource>(), stream, uiElement, isAutoSize, parentElement, imageContainer, isVisible);
+                    }
+                });
+        }
+        else
+        {
+            auto setSourceAction = imageSource.as<winrt::BitmapImage>().SetSourceAsync(stream);
+
+            setSourceAction.Completed([weakThis = this->get_weak(), uiElement, isAutoSize, parentElement, imageContainer, isVisible](
+                winrt::IAsyncAction const& /*operation*/, winrt::AsyncStatus status)
+                {
+                    if (status == winrt::AsyncStatus::Completed && isAutoSize)
+                    {
+                        if (auto strongThis = weakThis.get())
+                        {
+                            strongThis->SetAutoSize(uiElement, parentElement, imageContainer, isVisible, false /* imageFiresOpenEvent */);
+                        }
+                    }
+                });
+        }
+    }
+
+    winrt::IAsyncAction render_xaml::XamlBuilder::SetSvgUriSource(winrt::SvgImageSource image, winrt::Uri imageUrl)
+    {
+        try
+        {
+            co_await winrt::resume_foreground(image.Dispatcher());
+            image.as<winrt::SvgImageSource>().UriSource(imageUrl);
+        }
+        catch (winrt::hresult_wrong_thread ex)
+        {
+            // We didn't switch to the UI thread
```

Comment:
I don't think we need to fail fast here necessarily but logging the error would be nice :)

Reply:
Added an error statement and passed the `renderContext` to these methods